rmgpy/data/surface.py

```diff
@@ -0,0 +1,282 @@
+        points to the top-level folder of the solvation database.
+        
+        Load the metal library
+        """
+        self.libraries['surface'].load(os.path.join(path, 'libraries', 'metal.py'))
+
+    def get_binding_energies(self, metal_label):
+        """
+        Get a metal's binding energies from its label
+        """
+        return self.libraries['surface'].get_binding_energies(metal_label)
+
+    def get_surface_site_density(self, metal_label):
+        """
+        Get a metal's surface site desnity from its label
+        """
+        return self.libraries['surface'].get_surface_site_density(metal_label)
+
+    def get_all_entries_on_metal(self, metal):
+        """
+        Get all entries from the database that are on a certain metal, on any facet,
+        returning the labels.
+        """
+        return self.libraries['surface'].get_all_entries_on_metal(metal)
+
+    def find_binding_energies(self, metal):
+        """
+        Tries to find a match in the database when given a metal and/or facet and returns the binding energies or a
+        best guess of binding energies.
+        """
+        if metal is None:
+            raise DatabaseError("Cannot search for nothing.")
+        assert isinstance(metal, str)
+
+        facet = re.search('\d+', metal)
+
+        if facet is not None:
+            try:
+                metal_binding_energies = self.libraries['surface'].get_binding_energies(metal)
+            except DatabaseError:
+                # no exact match was found, so continue on as if no facet was given
+                logging.warning("Requested metal %r not found in database.", metal)
+                facet = None
+                metal = metal[:facet.span()[0]]
+                logging.warning("Searching for generic %r.", metal)
+
+        if facet is None:
+            # no facet was specified, so use the first
+            metal_entry_matches = self.libraries['surface'].get_all_entries_on_metal(metal)
+
+            if len(metal_entry_matches) == 1:
+                metal_binding_energies = self.libraries['surface'].get_binding_energies(metal_entry_matches[0])
+            elif metal_entry_matches is None:  # no matches
+                raise DatabaseError(f"No metal {metal} found in metal database.")
+            else:  # multiple matches
+                # average the binding energies together? just pick the first one?
+                # just picking the first one for now...
+                logging.warning(f"Found multiple binding energies for {metal!r}. Using {metal_entry_matches[0]!r}.")
+                metal_binding_energies = self.libraries['surface'].get_binding_energies(metal_entry_matches[0])
+
+        for element, energy in metal_binding_energies.items():
+            metal_binding_energies[element] = rmgpy.quantity.Energy(energy)
+            # ToDo: this conversion should be done once when loading the database, not every time we query it.
```

Comment:
this conversion should be done once when loading the database, not every time we query it.

Reply:
Resolved in bb7258ce83075e9526c62ad2ce353447f5e92d30